src/lsm/lsm_worker.c

```diff
@@ -93,11 +92,13 @@ __lsm_worker(void *arg)
 
 	cookie = (WT_LSM_WORKER_ARGS *)arg;
 	session = cookie->session;
-	conn = S2C(session);
 
 	entry = NULL;
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
```

Comment:
I'm personally a fan of having a pair of flags controlling thread shutdown. My concern is what happens if one of the manager threads stops due to an error, and skips clearing the sub-system flag. In that case shutdown will hang because the worker threads will never get the signal to stop.

Having said that..  other subsystems don't rely on a shared flag, so I don't have much of a leg to stand on here.